### PR TITLE
state machine generates a ctime

### PR DESCRIPTION
**Pull request by mmaxim**
@maxtaco @patrickxb @jacobhaven this change adds another return value to `ConsumeMessage` (the ctime) that will come along with `BroadcastMessage` so gregord and the client don't have two different ctime's for the same message.

### CODE REVIEW

File: rpc/server/main.go [hunk 1]
```diff
@@ -422,12 +429,15 @@ type syncReq struct {
...
-	retCh := make(chan error)
+func (s *Server) storageConsumeMessage(m gregor.Message) consumeMessageRet {
+	retCh := make(chan consumeMessageRet)
 	req := consumeMessageReq{m: m, respCh: retCh}
 	err := s.storageDispatch(req)
 	if err != nil {
-		return err
+		return consumeMessageRet{
+			err:   err,
+			ctime: time.Now(),
```
Comment (patrickxb) [inline rpc/server/main.go]: Can you remove this line in the case of an error?

**Comment by patrickxb** [comment]
LGTM.  one question, but not important.

File: rpc/server/main.go (hunk 1 above)
Comment (mmaxim) [inline rpc/server/main.go]: Sure!

**Comment by mmaxim** [comment]
@maxtaco can I merge this?

**Comment by maxtaco** [comment]
👍

**Pull request merged**